### PR TITLE
Ignore stdout/stderr broken pipe errors

### LINKED ISSUE #1116: BrokenPipeError leading to a crash when using filtering/counting shell utilities
When using e.g. tail or head to limit the number of lines a BrokenPipeError is thrown and leads to an abnormal exit of borg:

```
% sudo borg list BACKUP | head -n1
20160202-121810                      Tue, 2016-02-02 13:03:51
Local Exception.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 1498, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 1452, in run
    return args.func(args)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 80, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 544, in do_list
    print(format_archive(archive_info))
BrokenPipeError: [Errno 32] Broken pipe

Platform: Linux occam 4.6.0-bait.0-wbt #1 SMP PREEMPT Tue May 17 09:32:06 CEST 2016 x86_64 
Linux: debian stretch/sid 
Borg: 1.0.2  Python: CPython 3.5.1+
PID: 21174  CWD: /home/user
sys.argv: ['/usr/bin/borg', 'list', 'BACKUP']
SSH_ORIGINAL_COMMAND: None

Exception ignored in: <_io.TextIOWrapper name='<stdout>' encoding='UTF-8'>
BrokenPipeError: [Errno 32] Broken pipe
```

***

:moneybag: [there is a bounty for this](https://www.bountysource.com/issues/34723987-brokenpipeerror-leading-to-a-crash-when-using-filtering-counting-shell-utilities)

### PR DESCRIPTION
**Pull request by PlasmaPower**
Redo of #1375 for 1.0-maint

Fixes #1116

### CODE REVIEW

File: borg/helpers.py [hunk 1]
```diff
@@ -1089,3 +1091,31 @@ def log_multi(*msgs, level=logging.INFO):
...
+
+
+class ErrorIgnoringTextIOWrapper(io.TextIOWrapper):
+    def read(self, n):
+        if not self.closed:
+            try:
+                return super(ErrorIgnoringTextIOWrapper, self).read(n)
+            except OSError as err:
+                if err.errno != errno.EPIPE:
+                    raise
```
Comment (enkore) [inline borg/helpers.py]: except BrokenPipeError?

Comment (PlasmaPower) [inline borg/helpers.py]: That doesn't exist for whatever reason, and I can't find any mention of it in the docs. I think it's just how the error prints out maybe?

Comment (PlasmaPower) [inline borg/helpers.py]: Or not, I think I might have just made a typo. Fixed in just a sec.

File: borg/helpers.py [hunk 2]
```diff
@@ -1089,3 +1091,27 @@ def log_multi(*msgs, level=logging.INFO):
         lines.extend(msg.splitlines())
     for line in lines:
         logger.log(level, line)
+
+
+class ErrorIgnoringTextIOWrapper(io.TextIOWrapper):
+    def read(self, n):
+        if not self.closed:
+            try:
+                return super(ErrorIgnoringTextIOWrapper, self).read(n)
```
Comment (ThomasWaldmann) [inline borg/helpers.py]: python3 has parameter-less super():  return super().read(n)

Comment (PlasmaPower) [inline borg/helpers.py]: Oh, sweet! Fixed.

**Comment by enkore** [comment]
lgtm

**Pull request merged**